### ISSUE TITLE
Virtual Kubelet Node is not capable of running pods throwing "DuplicateVolumes": Duplicate volumes 'kube-api-access-sndws' are found

### ISSUE THREAD

**Issue by tmcmm** [virtual-kubelet/azure-aci#303]
Since yesterday applicational pods stopped running throwing the following error:
```
Normal Scheduled 6m55s default-scheduler Successfully assigned xxxxx/test-e5ccbaf3-9xxxx5723ac2bc45-66654b7f99-f8t9c to virtual-node-aci-linux
Warning ProviderCreateFailed 75s (x16 over 6m55s) virtual-node-aci-linux/pod-controller api call to  https://management.azure.com/subscriptions/xxxxx/resourceGroups/MC_xxxxxx-aks-prod_westeurope/providers/Microsoft.ContainerInstance/containerGroups/xxxxxx?api-version=2021-07-01: got HTTP response status code 400 error code "DuplicateVolumes": Duplicate volumes 'kube-api-access-sndws' are found.
```

**Explanation:**
By default, a Pod is associated with a service account, and a credential (token) for that service account is placed into the filesystem tree of each container in that Pod, at /var/run/secrets/kubernetes.io/serviceaccount/token.
If available, a certificate bundle is placed into the filesystem tree of each container at /var/run/secrets/kubernetes.io/serviceaccount/ca.crt, and should be used to verify the serving certificate of the API server.
Finally, the default namespace to be used for namespaced API operations is placed in a file at /var/run/secrets/kubernetes.io/serviceaccount/namespace in each container.

Tried to redeploy the application on the same namespace.
Tried to redeploy the application on a new namespace.
Tried to re-enable the virtual node addon.

From the logs of aci-connector:
kubectl logs -l app=aci-connector-linux -n kube-system
```
time="2022-09-27T12:27:43Z" level=info msg="Event(v1.ObjectReference{Kind:\"Pod\", Namespace:\"test\", Name:\"sxxxxxx9-jhq25\", UID:\"237973dd-dec6-4c86-bef6-d1cb8eb009af\", APIVersion:\"v1\", ResourceVersion:\"\", FieldPath:\"\"}): type: 'Warning' reason: '**ProviderCreateFailed**' api call to https://management.azure.com/subscriptions/xxxxxx/resourceGroup/sxxxx-prod_westeurope/providers/Microsoft.ContainerInstance/containerGroups/xxxxxx?api-version=2021-07-01: got HTTP response status code 400 error code \"DuplicateVolumes\":  Duplicate volumes 'kube-api-access-sndws' are found.." node=virtual-node-aci-linux operatingSystem=Linux provider=azure watchedNamespace=
```

Sample pod gets created and running
I have tried to create a sample pod on the same namespace and to test if it was a problem of the virtual kubelet but this new pod managed to run:

```
cat << EOF | kubectl apply -f-
apiVersion: v1
kind: Pod
metadata:
  labels:
    run: nginx
  name: nginx
  namespace: test
spec:
  containers:
  - image: nginx
    name: nginx
    resources: {}
  nodeSelector:
    kubernetes.io/role: agent
    beta.kubernetes.io/os: linux
    type: virtual-kubelet
  tolerations:
  - key: virtual-kubelet.io/provider
    operator: Exists
  - key: azure.com/aci
    effect: NoSchedule
  restartPolicy: Never
EOF
```
kubectl get pods -n test
NAME READY STATUS RESTARTS AGE
nginx 1/1 Running 0 79s
 

![image](https://user-images.githubusercontent.com/16254825/192557101-58978dd4-b828-411d-bdfc-6bfcd0e50b18.png)

The only thing that seems to fix is adding the following on deployment spec:
```
automountServiceAccountToken: false
```

**Comment by asankaf** [comment]
I have also noticed that as of the current version, we can't seem to have multiple volume mounts. I tried mounting one EmptyDir volume for logging purposes and one Azure File Share volume for storage purposes, but it throws the same weird "DuplicateVolume" ProviderFailed exception during the deployment.

**Comment by helayoty** [comment]
@tmcmm @asankaf Thanks for reaching out, we are preparing a hotfix to solve this issue asap.

**Comment by ayelencasamassa** [comment]
OMG! i thought i was crazy since I'm having the same issue since monday and I can't find any solution apart from disabling all volumes mount on my deployment. Please, find a solution soon :(

**Comment by helayoty** [comment]
Kindly refer to the downgrade steps [here](https://github.com/virtual-kubelet/azure-aci/blob/master/docs/DOWNGRADE-README.md) to go back to Release 1.4.1 until the fix rolls out.

**Comment by ayelencasamassa** [comment]
> Kindly refer to the downgrade steps [here](https://github.com/virtual-kubelet/azure-aci/blob/master/docs/DOWNGRADE-README.md) to go back to Release 1.4.1 until the fix rolls out.

That step-by-step didn't work well... Anyway, I managed to downgrade the ACI version to 1.4.1 in my cluster and the issue is gone. Please, let us know when version 1.4.4 or newer is ready.

**Comment by helayoty** [comment]
> > Kindly refer to the downgrade steps [here](https://github.com/virtual-kubelet/azure-aci/blob/master/docs/DOWNGRADE-README.md) to go back to Release 1.4.1 until the fix rolls out.
> 
> Downgrading does not work for some reason. Want to get VIRTUALNODE_USER_IDENTITY_CLIENTID, but it is empty in aci-connector-linux-XXXX pod.

As mentioned in the steps, `the client id of the aciconnectorlinux-XXXX managed identity in the MC_XXXX resource group.`

**Comment by nakazonot** [comment]
> As mentioned in the steps, `the client id of the aciconnectorlinux-XXXX managed identity in the MC_XXXX resource group.`

understood. It means that there is aciconnectorlinux-XXXX in the MC_XXXX resource group.
However, there was no aciconnectorlinux-XXXX. (There is aks-agentpool-XXXX-nsg and aks-nodepool1-XXXX-vmss.)

It's strange that virtual-node-aci-linux is enabled.

**Comment by helayoty** [comment]
> > > As mentioned in the steps, `the client id of the aciconnectorlinux-XXXX managed identity in the MC_XXXX resource group.`
> > 
> > 
> > understood. It means that there is aciconnectorlinux-XXXX in the MC_XXXX resource group. However, there was no aciconnectorlinux-XXXX. (There is aks-agentpool-XXXX-nsg and aks-nodepool1-XXXX-vmss.)
> > It's strange that virtual-node-aci-linux is enabled.
> 
> It depends on how you enable virtual-node. For example, I did it by enabling the addon using as cli. I didn't use helm. Az cli creates everything you need for it but, in my case, I couldn't downgrade the version. I tried but then aks automaticaly upgrade the version again. So, that is not a good solution, we need the fix.

You'll need to give the downgrade node another name than the one created by AKS to prevent the auto-update.

**Comment by ayelencasamassa** [comment]
> @tmcmm @ayelencasamassa @nakazonot the fix (v1.4.5) has been rolled out on AKS now.

checked! seems like it's working.. at least the pod is alive.
I will test the functionality.
Thanks @helayoty, you rock it!

**Comment by nakazonot** [comment]
@helayoty @ayelencasamassa Thanks! Upgraded and working fine now.
